pages/[league]/player/[id].tsx

```diff
@@ -109,6 +146,32 @@ export default ({ playerId, league }: { playerId: string; league: League }) => {
     enabled: !!playerTypeInfo?.playerType,
   });
 
+  useEffect(() => {
+    let spinnerHeight;
+    let statsHeight;
+
+    if (
+      !isIndexView &&
+      (playerInfoLoading || playerRatingsLoading || playerStatsLoading)
+    ) {
+      spinnerHeight = document.getElementById('player-loading')?.offsetHeight;
+    }
+
+    if (!isIndexView && (playerInfo || playerRatings || playerStats)) {
+      statsHeight = document.getElementById('player-details')?.offsetHeight;
+    }
+
+    parent.postMessage(spinnerHeight ? spinnerHeight : statsHeight ?? 0, '*');
+  }, [
+    isIndexView,
+    playerInfo,
+    playerInfoLoading,
+    playerRatings,
+    playerRatingsLoading,
+    playerStats,
+    playerStatsLoading,
+  ]);
```

Comment:
A couple more suggestions here:
1. We should return early if the user is in the index view since they don't need to run this code
2. We should store the loading values in a separate variable (and share that with the condition for the loader) so we can reduce duplication and make things more readable.
3. We should be using refs to get the values instead of the `document.getElementById` method. This is a bit tricker but will make things work better so let me know if you have any questions. 

```suggestion
  const isLoading = !playerInfo || !playerRatings || !playerStats;
  const loaderRef = useRef(); // connect this ref to the loader instead of passing an ID
  const detailsRef = useRef(); // connect this ref to the loader instead of passing an ID
  useEffect(() => {
    if (shouldShowIndexView)  return;
    
    const contentHeight = isLoading ? loaderRef.current?.offsetHeight : detailsRef.current?.offsetHeight;

    parent.postMessage(contentHeight ?? 0, '*');
  }, [
    isLoading, shouldShowIndexView
  ]);
```

If you want to pair to better understand any of this feel free to ask I think the ref stuff may come to be a bit tricky